core/src/main/scala/cats/data/Chain.scala

```diff
@@ -1416,3 +1412,13 @@ private[data] trait ChainPartialOrder[A] extends PartialOrder[Chain[A]] {
 
   override def eqv(x: Chain[A], y: Chain[A]): Boolean = x === y
 }
+
+private[data] object ChainMonoid {
+  private val singleton: Monoid[Chain[Any]] = new Monoid[Chain[Any]] {
+    def empty: Chain[Any] = Chain.nil
```

Comment:
Do we really need an object for this? can we just put it alongside the `implicit def`.

Reply:
Maybe not - I did it for consistency with the other cases but happy to change it.

That also raises the question of whether to go ahead with adding the companion objects in the other cases. What do you think?

Reply:
That's a good question. In those cases the monoid class was publicly exposed. In this case the object is private, seems to be created unnecessary indirection

Tbh I just don't think it's that valuable—if users directly want the monoid instance can't they just grab it from the `implicit def instances` or whatever?

Reply:
I think you run into bincompat issues if you put it on the trait. Or at least I did in https://github.com/typelevel/cats/pull/4313

Reply:
You are right, but should be fine if either the trait is private/sealed or it's actually an (abstract) class 😛

Reply:
Ah gotcha 👍 Whereas `ListInstances` is a public trait